### ISSUE TITLE
Create a QGIS plugin for managing and deploying QWC2 configurations

### ISSUE THREAD

**Issue by andreasneumann** [qgis/qwc2-demo-app#3]
The idea is to create a QGIS plugin that:

1. can edit configuration files
2. can deploy to a test or production system

It should be able to edit at least:
- config.json
- themesConfig.json

Maybe more files later.

**Comment by d3netxer** [comment]
does anybody have a good guide on how to start using QGIS, QGIS Server, and QWC2? 

There are good presentations online on the great features and capabilities of these components but I have a hard time finding guides on how to use them in practice.

![screenshot from 2017-12-28 10-28-54](https://user-images.githubusercontent.com/2665840/34414940-08838a96-ebba-11e7-813d-db871bbe322d.png)

I would be willing to help with documentation.

So to recap I am interested in getting from QGIS desktop and creating a QGIS project, then taking that project an uploading to QGIS server. Then QGIS server would be able to display the project using QWC2 on the Internet.

**Comment by tudorbarascu** [comment]
For QGIS Desktop + Server part should be enough:
* https://docs.qgis.org/testing/en/docs/training_manual/qgis_server/index.html
* https://docs.qgis.org/testing/en/docs/user_manual/working_with_ogc/index.html

For the QWC2 part you need to edit the configuration files manually, but as you already have some examples that shouldn't be too hard.
That being said, your request should have been a new issue.
Good luck.

**Comment by d3netxer** [comment]
Ok fair enough @tudorbarascu I created a new issue #70 

I will take a look at your links now!

**Comment by FlavBd** [comment]
Hello, 

like we explain in the [mailing list](http://osgeo-org.1560.x6.nabble.com/New-features-in-QWC2-td5434695.html) we are working on a QGIS plugin to add new project directly from QGIS desktop to a qwc2 application. I proposed something in [this repo](https://github.com/FlavBd/qwc2_plugin).

Everything is explain in the README but the idea is to create a new micro-service build with a qwc2 app. The user will download the plugin and add it in QGIS to publish easily a new project.  

We are now working on authentication part to create new themes.json

Feel free to add comments or/and suggest changes.

**Comment by pka** [comment]
Great news, @FlavBd! I will have a look at it (probably after my holidays). cc @HusseinKabbout

**Comment by FlavBd** [comment]
Hello !

I come back to you because we have developed a microservice for our plugin which is available on [this GitHub repository](https://github.com/FlavBd/qwc2_plugin). In the readme you have a link to [a video](https://www.youtube.com/watch?v=XK5nevc4D9g&feature=youtu.be) which show you how it works. 

What do you think about it ? Do you think it can be added to the [microservices repo](https://github.com/qwc-services) ?

**Comment by pka** [comment]
Watched the video, looks nice :+1:!

We're generally open in adding more repos to the qwc-services organization. We have repos with different prefixes, which categorize them into officially supported services (`qwc_*`), custom services (`sogis_*`) and others. The QGIS plugin would be in a new category. @HusseinKabbout has written a similar plugin, but we've never advertized it, because we're still not sure, which way of publishing and configuring the majority of QWC2 users prefer.

I have an other proposal for showcasing your plugin: an "awesome QWC" list with links and short descriptions of available microservices and tools like yours. What do you think?

**Comment by ar-siddiqui** [comment]
> Hello,
> 
> like we explain in the [mailing list](http://osgeo-org.1560.x6.nabble.com/New-features-in-QWC2-td5434695.html) we are working on a QGIS plugin to add new project directly from QGIS desktop to a qwc2 application. I proposed something in [this repo](https://github.com/FlavBd/qwc2_plugin).
> 
> Everything is explain in the README but the idea is to create a new micro-service build with a qwc2 app. The user will download the plugin and add it in QGIS to publish easily a new project.
> 
> We are now working on authentication part to create new themes.json
> 
> Feel free to add comments or/and suggest changes.

Has this plugin been published? And is it something similar to QGIS Cloud? because in your video the interface looks almost the same as QGIS Cloud Webmaps.
